### PR TITLE
DN-621: parsing 'timestamp' key

### PR DESCRIPTION
**Pull request by juliodealroom**
We want to be able to use timestamps in firestore. That is why in order to use the firestore-service, we need to check if a key 'timestamp' is passed. In this case, we add the `firestore.SERVER_TIMESTAMP` as its value.

### CODE REVIEW

File: dealroom_firestore_connector/__init__.py [hunk 1]
```diff
@@ -58,8 +58,27 @@ def get(doc_ref, *args, **kwargs):
...
+    """
+    for k, v in obj.items():
+      if isinstance(v, dict):
+        obj[k] = _parse_payload(v, key, replace_value)
+      if isinstance(v, list):  
+        obj[k] = [_parse_payload(i) for i in v]
+      if key in obj:
+        obj[key] = replace_value
+    return obj
+
```
Comment (juliodealroom) [inline dealroom_firestore_connector/__init__.py:79]: It searches recursively for the key passed as parameter, even if the data contains dictionaries or lists

File: dealroom_firestore_connector/__init__.py [hunk 2]
```diff
@@ -275,8 +294,11 @@ def set_history_doc_refs(
     """
 
     history_col = db.collection(HISTORY_COLLECTION_PATH)
+    
+    # Setting SERVER_TIMESTAMP for 'timestamp' key
+    parsed_payload = _parse_payload(payload, 'timestamp', firestore.SERVER_TIMESTAMP)
```
Comment (juliodealroom) [inline dealroom_firestore_connector/__init__.py:299]: use parsed_payload variable instead, which is the result of the parsing of the original payload

Comment (saltas888) [reply dealroom_firestore_connector/__init__.py:299]: All these versions of payload got confused me. I would start here by defining the `_payload` variable and would continue with this in the following transformations

File: dealroom_firestore_connector/__init__.py [hunk 3]
```diff
@@ -275,8 +294,11 @@ def set_history_doc_refs(
...
 
-    _payload = {**payload}
+    _parsed_payload  = {**parsed_payload}
```
Comment (saltas888) [inline dealroom_firestore_connector/__init__.py:301]: this won't be needed cause you 'll have already created a clone of it ⬆️

File: dealroom_firestore_connector/__init__.py [hunk 4]
```diff
@@ -292,14 +314,14 @@ def set_history_doc_refs(
     # CREATE: If there are not available documents in history
     elif len(history_refs) == 0:
         # Add any default values to the payload
-        _payload = {
+        _parsed_payload  = {
```
Comment (saltas888) [inline dealroom_firestore_connector/__init__.py:317]: would leave it as is

File: dealroom_firestore_connector/__init__.py [hunk 5]
```diff
@@ -292,14 +314,14 @@ def set_history_doc_refs(
...
             "dealroom_id": _NOT_IN_DEALROOM_ENTITY_ID,
             "final_url": "",
-            **payload,
+            **parsed_payload,
```
Comment (saltas888) [inline dealroom_firestore_connector/__init__.py:320]: would use _payload, which means that we assign to _payload itself + some default vars if needed

File: dealroom_firestore_connector/__init__.py [hunk 6]
```diff
@@ -292,14 +314,14 @@ def set_history_doc_refs(
...
         }
         # Validate that the new document will have the minimum required fields
         try:
-            _validate_new_history_doc_payload(_payload)
+            _validate_new_history_doc_payload(_parsed_payload)
```
Comment (saltas888) [inline dealroom_firestore_connector/__init__.py:324]: same

**Pull request closed without merging**